File: practical.js
function isLeapYear(year){
    return(year % 4 === 0 && year % 100 !==0) || (year % 400 === 0);
}

function getDaysInMonth(month, year){
    switch(month){
        case 1: return 31;
        case 2: return isLeapYear(year) ? 29 : 28;
        case 3: return 31;
        case 4: return 30;
        case 5: return 31;
        case 6: return 30;
        case 7: return 31;
        case 8: return 31;
        case 9: return 30;
        case 10: return 31;
        case 11: return 30;
        case 12: return 31;
        default: return 0;
    }
}

function generateDatesForYear(year){
    let dates = [];
    for(let month = 1; month <= 12; month++){
        let daysInonth = getDaysInMonth(month, year);
        for(let day = 1; day <= daysInonth; day++){
            let dd = day.toString().padStart(2, '0');
            let mm = month.toString().padStart(2, '0');
            dates.push(`${dd}/${mm}/${year}`);
    }
    return dates;
}
}

let year = 2024;
console.log(generateDatesForYear(year));

File: practical.c
#include <stdio.h>
#include <stdbool.h>

// Function to check if a year is a leap year
bool isLeapYear(int year) {
    if (year % 4 == 0) {
        if (year % 100 == 0) {
            if (year % 400 == 0)
                return true;
            else
                return false;
        } else {
            return true;
        }
    } else {
        return false;
    }
}

// Function to validate the date
bool isValidDate(int day, int month, int year) {
    // Check if the year is valid
    if (year < 1)
        return false;

    // Check if the month is valid
    if (month < 1 || month > 12)
        return false;

    // Days in each month
    int daysInMonth[] = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };

    // Adjust for leap year in February
    if (isLeapYear(year)) {
        daysInMonth[1] = 29;
    }

    // Check if the day is valid
    if (day < 1 || day > daysInMonth[month - 1])
        return false;

    return true;
}

// Test cases for date validation
void testDateValidation() {
    // Test valid dates
    printf("Test 1: %s\n", isValidDate(15, 8, 2024) ? "Passed" : "Failed"); // Valid date
    printf("Test 2: %s\n", isValidDate(29, 2, 2024) ? "Passed" : "Failed"); // Valid leap year date
    printf("Test 3: %s\n", isValidDate(31, 12, 2024) ? "Passed" : "Failed"); // Valid date

    // Test invalid dates
    printf("Test 4: %s\n", isValidDate(0, 8, 2024) ? "Passed" : "Failed"); // Invalid day
    printf("Test 5: %s\n", isValidDate(15, 13, 2024) ? "Passed" : "Failed"); // Invalid month
    printf("Test 6: %s\n", isValidDate(29, 2, 2023) ? "Passed" : "Failed"); // Invalid leap year date (not a leap year)
    printf("Test 7: %s\n", isValidDate(32, 1, 2024) ? "Passed" : "Failed"); // Invalid day
}

// Main function to execute test cases
int main() {
    testDateValidation();
    return 0;
}